src/event.ts

```diff
@@ -17,6 +17,12 @@ export type Event = {
   calendar: string;
 };
 
+export type EventFormatterFunction = (event: Event) => string;
+
+const customEventFormatterWrapper = (formatter: EventFormatterFunction, event:Event, defaultValue: string): string => {
```

Comment:
I will answer both comments about this function signature here:
* I chose to pass in the custom formatter for two reasons: 1 is that the use of a global as a necessary evil so that users could provide it in a separate JS block, but did not want to bury it deep but instead have it as outside as I could, to decouple most internals from the fact that it is provided as a global. 2. It makes testing cleaner and easier
* I chose to pass the tag to the function also to decouple the default replacement from the actual tag that is chosen for the custom formatter. In this case, if we don't get the tag, we'd need to hardcode it within this function too, or maybe have it as a top-level constant, but I thought that was too far from the pattern of chained replacements below, which reads very well as its own documentation.